### PR TITLE
Add energy computation to RadarPostProcessNode

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

File: test/src/scene/rcsAngleDistributionTest.cpp [hunk 1]
```diff
@@ -12,25 +12,143 @@
...
+	const auto refPolU = -1.0f * polU;
+	const auto refPolR = rayDir.cross(refPolU);
+
+	const auto polCompU = pol.dot(polU);
+	const auto polCompR = pol.dot(polR);
+
+	return -polCompR * refPolR + polCompU * refPolU;
+}
+
+float getRayArea(float R) { return R * R * std::sin(elevationRad) * azimuthRad; }
```
Comment (PawelLiberadzki) [inline test/src/scene/rcsAngleDistributionTest.cpp:55]:
```suggestion
float getRayArea(float R) { return R * R * std::sin(elevationStep) * azimuthStep; }
```

File: src/RGLFields.hpp [hunk 2]
```diff
@@ -55,34 +56,34 @@ typedef unsigned char TextureTexelFormat;
...
-	    RELATIVE_VELOCITY_VEC3_F32,
-	    RADIAL_SPEED_F32,
-	    POWER_F32,
-	    RCS_F32,
-	    NOISE_F32,
-	    SNR_F32,
-	    NORMAL_VEC3_F32,
-	    INCIDENT_ANGLE_F32,
-	};
+	static std::set<rgl_field_t> allRealFields = {XYZ_VEC3_F32,
```
[the remark this answers is not in the document]
Comment (prybicki) [reply src/RGLFields.hpp:66]: Both are the result of our clang-format. The new one misses the colon at the end of list, which causes different formatting. Fixed that.

File: src/gpu/nodeKernels.cu [hunk 3]
```diff
@@ -78,6 +79,77 @@ __global__ void kFilter(size_t count, const Field<RAY_IDX_U32>::type* indices, c
...
+	constexpr float reflectionCoef = 1.0f; // TODO
+	const thrust::complex<float> i = {0, 1.0};
+	const Vec3f dirX = {1, 0, 0};
+	const Vec3f dirY = {0, 1, 0};
+	const Vec3f dirZ = {0, 0, 1};
+
+	Vec3f rayDirCts = rayPose[tid] * Vec3f{0, 0, 1};
+	Vec3f rayDirSph = {rayDirCts.length(), rayDirCts[0] == 0 && rayDirCts[1] == 0 ? 0 : atan2(rayDirCts.y(), rayDirCts.x()),
+	                   std::acos(rayDirCts.z() / rayDirCts.length())};
+	float phi = rayDirSph[1]; // azimuth, 0 = X-axis, positive = CCW
```
[the remark this answers is not in the document]
Comment (prybicki) [reply src/gpu/nodeKernels.cu:120]: Added const where possible.

File: src/gpu/nodeKernels.cu [hunk 4]
```diff
@@ -78,6 +79,77 @@ __global__ void kFilter(size_t count, const Field<RAY_IDX_U32>::type* indices, c
...
+
+	Vector<3, thrust::complex<float>> rayPolCplx = {reflectedPol.x(), reflectedPol.y(), reflectedPol.z()};
+
+	Vector<3, thrust::complex<float>> apE = reflectionCoef * exp(i * kr) * rayPolCplx;
+	Vector<3, thrust::complex<float>> apH = -apE.cross(rayDir);
+
+	Vec3f vecK = waveNum * ((dirX * cp + dirY * sp) * st + dirZ * ct);
+
+	float rayArea = hitDist[tid] * hitDist[tid] * std::sin(rayElevationStepRad) * rayAzimuthStepRad;
+	Vector<3, thrust::complex<float>> outBUBRFactor = {0, 0, 0};
```
[the remark this answers is not in the document]
Comment (prybicki) [reply src/gpu/nodeKernels.cu:146]: Fixed

File: test/src/scene/rcsAngleDistributionTest.cpp (hunk 1 above)
Comment (prybicki) [reply test/src/scene/rcsAngleDistributionTest.cpp]: Fixed manually

**Review by msz-rai — APPROVED** [review]

**Pull request merged**